spec/kintsugi_integration_spec.rb

```diff
@@ -0,0 +1,149 @@
+      git.add(all: true)
+      git.commit("Add target bar")
+
+      `git -C #{git_directory_path} #{git_command} #{first_commit_hash} &> /dev/null`
+      Kintsugi.run([File.join(project.path, "project.pbxproj")])
+
+      project = Xcodeproj::Project.open(project.path)
+      expect(project.targets.map(&:display_name)).to contain_exactly("foo", "bar")
+    end
+
+    it "resolves conflicts automatically with driver" do
+      git.config("merge.kintsugi.name", "Kintsugi driver")
+      git.config("merge.kintsugi.driver", "#{__dir__}/../bin/kintsugi driver %O %A %B %P")
+      File.write(File.join(git_directory_path, ".gitattributes"), "*.pbxproj merge=kintsugi")
+
+      project = create_new_project_at_path(File.join(git_directory_path, project_name))
+
+      git.add(File.join(git_directory_path, ".gitattributes"))
+      git.add(project.path)
+      git.commit("Initial project")
+
+      project.new_target("com.apple.product-type.library.static", "foo", :ios)
+      project.save
+
+      git.add(all: true)
+      git.commit("Add target foo")
+      first_commit_hash = git.revparse("HEAD")
+
+      git.checkout("HEAD^")
+      project = Xcodeproj::Project.open(project.path)
+      project.new_target("com.apple.product-type.library.static", "bar", :ios)
+      project.save
+      git.add(all: true)
+      git.commit("Add target bar")
+
+      `git -C #{git_directory_path} #{git_command} #{first_commit_hash} &> /dev/null`
+
+      project = Xcodeproj::Project.open(project.path)
+      expect(project.targets.map(&:display_name)).to contain_exactly("foo", "bar")
+    end
+
+    it "keeps conflicts if failed to resolve conflicts" do
+      File.write(File.join(git_directory_path, ".gitattributes"), "*.pbxproj merge=Unset")
+
+      project = create_new_project_at_path(File.join(git_directory_path, project_name))
+      project.new_target("com.apple.product-type.library.static", "foo", :ios)
+      project.save
+
+      git.add(File.join(git_directory_path, ".gitattributes"))
+      git.add(project.path)
+      git.commit("Initial project")
+
+      project.targets[0].build_configurations.each do |configuration|
+        configuration.build_settings["PRODUCT_NAME"] = "bar"
+      end
+      project.save
+      git.add(all: true)
+      git.commit("Change target product name to bar")
+      first_commit_hash = git.revparse("HEAD")
+
+      git.checkout("HEAD^")
+      project = Xcodeproj::Project.open(project.path)
+      project.targets[0].build_configurations.each do |configuration|
+        configuration.build_settings["PRODUCT_NAME"] = "baz"
+      end
+      project.save
+      git.add(all: true)
+      git.commit("Change target product name to baz")
+
+      `git -C #{git_directory_path} #{git_command} #{first_commit_hash} &> /dev/null`
+
+      expect {
+        Kintsugi.run([File.join(project.path, "project.pbxproj")])
+      }.to raise_error(Kintsugi::MergeError)
+      expect {
+        project = Xcodeproj::Project.open(project.path)
```

Comment:
is this the best way to check if conflicts are kept?
Maybe just diff the file with the previous version?